src/router.js

```diff
@@ -70,16 +57,45 @@ export class Router {
     return this;
   }
 
+  createRootedPath(fragment) {
+    let path = '';
+
+    if (this.baseUrl.length && this.baseUrl[0] !== '/') {
+      path += '/';
+    }
+
+    path += this.baseUrl;
+
+    if (path[path.length - 1] != '/' && fragment[0] != '/') {
+      path += '/';
+    }
+
+    return normalizeAbsolutePath(path + fragment, this.history._hasPushState);
```

Comment:
We should add a `getRoot()` method to `History` so this code can stop relying on private data.